[quant][graphmode][fx] Add functional conv2d + relu

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51155 [quant][graphmode][fx] Add support for functional conv1d and conv3d
* **#51079 [quant][graphmode][fx] Add functional conv2d + relu**

Summary:
Added support for functional conv2d + relu, will add conv1d and conv3d in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_functional_conv

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26089964](https://our.internmc.facebook.com/intern/diff/D26089964)

jerryzh168 committed:
[quant][graphmode][fx] Add functional conv2d + relu

Summary:
Added support for functional conv2d + relu, will add conv1d and conv3d in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_functional_conv

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add functional conv2d + relu"


Summary:
Added support for functional conv2d + relu, will add conv1d and conv3d in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_functional_conv

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add functional conv2d + relu"


Summary:
Added support for functional conv2d + relu, will add conv1d and conv3d in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_functional_conv

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -218,12 +218,33 @@ def __init__(self, quantizer: QuantizerCls, node: Node):
+        # Supported combinations are:
+        # quant_type | activation (compute_type) | weight
+        #  static       quint8                      qint8
+        # tuple (activation_dtype, weight_dtype, compute_dtype)
```
is L224 related to L221:L223?  It was a bit confusing to read, I might be missing something

jerryzh168 replied:
yeah this is a comment for supported_dtypes

vkuzo commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -218,12 +218,33 @@ def __init__(self, quantizer: QuantizerCls, node: Node):
+        # quant_type | activation (compute_type) | weight
+        #  static       quint8                      qint8
+        # tuple (activation_dtype, weight_dtype, compute_dtype)
+        supported_dtypes = [
```
is this related to the comments above?

jerryzh168 replied:
yes

vkuzo commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -218,12 +218,33 @@ def __init__(self, quantizer: QuantizerCls, node: Node):
         # TODO: debug option for conv module
         qconfig = quantizer.qconfig_map[node.name]
+        dtypes = get_qconfig_dtypes(qconfig)
+        if dtypes not in supported_dtypes:
+            warnings.warn(
+                "dtype combination: {} is not "
+                "supported by Conv "
+                "supported dtype combinations are: {}".format(dtypes, supported_dtypes))
+            if self.relu_node:
```
maybe we can add a comment that this is going to leave the unsupported dtype pattern unquantized? just for easier reading of the code

jerryzh168 replied:
sure

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add functional conv2d + relu"


Summary:
Added support for functional conv2d + relu, will add conv1d and conv3d in future PR

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_functional_conv

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26089964](https://our.internmc.facebook.com/intern/diff/D26089964)

[ghstack-poisoned]
